src/charm.py

```diff
@@ -140,14 +140,49 @@ def _check_leader(self):
             self.logger.info("Not a leader, skipping setup")
             raise ErrorWithStatus("Waiting for leadership", WaitingStatus)
 
-    def _deploy_k8s_resources(self) -> None:
-        """Deploys K8S resources."""
+    def _check_and_report_k8s_conflict(self, error):
+        """Check for conflict and report it.
```

Comment:
This is okay, but can definitely be shortened to `"""Returns True if error status code is 409 (conflict on apply), False otherwise."""

Reply:
Will do.